controllers/gorch/guardrailsorchestrator_controller.go

```diff
@@ -0,0 +1,181 @@
+	return (&GuardrailsOrchestratorReconciler{
+		Client:    mgr.GetClient(),
+		Scheme:    mgr.GetScheme(),
+		Namespace: ns,
+		Recorder:  recorder,
+	}).SetupWithManager(mgr)
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the GuardrailsOrchestrator object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.1/pkg/reconcile
+func (r *GuardrailsOrchestratorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	// TODO(user): your logic here
+	orchestrator := &gorchv1alpha1.GuardrailsOrchestrator{}
+
+	err := r.Get(ctx, req.NamespacedName, orchestrator)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("GuardrailsOrchestrator resource not found. Ignoring since object must be deleted.")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get GuardrailsOrchestrator")
+		return ctrl.Result{}, err
+	}
+	if !controllerutil.ContainsFinalizer(orchestrator, finalizerName) {
+		log.Info("Adding Finalizer for the GuardrailsOrchestrator")
+		if ok := controllerutil.AddFinalizer(orchestrator, finalizerName); !ok {
+			return ctrl.Result{Requeue: true}, nil
+		}
+	}
+
+	if err = r.Update(ctx, orchestrator); err != nil {
+		log.Error(err, "Failed to update GuardrailsOrchestrator with finalizer")
+		return ctrl.Result{}, err
+	}
+
+	// TO-DO: handle deletion of the orchestrator
+	if controllerutil.ContainsFinalizer(orchestrator, finalizerName) {
+		controllerutil.RemoveFinalizer(orchestrator, finalizerName)
+		if err = r.Update(ctx, orchestrator); err != nil {
+			log.Error(err, "Failed to update GuardrailsOrchestrator without finalizer")
+			return ctrl.Result{}, err
+		}
+		log.Info("Succesfully removed finalizer from GuardrailsOrchestrator")
+	}
+
+	existingServiceAccount := &corev1.ServiceAccount{}
+	err = r.Get(ctx, types.NamespacedName{Name: orchestrator.Name, Namespace: orchestrator.Namespace}, existingServiceAccount)
+	if err != nil && errors.IsNotFound(err) {
+		serviceAccount := r.createServiceAccount(ctx, orchestrator)
+		log.Info("Creating a new ServiceAccount", "ServiceAccount.Namespace", serviceAccount.Namespace, "ServiceAccount.Name", serviceAccount.Name)
+		err = r.Create(ctx, serviceAccount)
+		if err != nil {
+			log.Error(err, "Failed to create new ServiceAccount", "ServiceAccount.Namespace", serviceAccount.Namespace, "ServiceAccount.Name", serviceAccount.Name)
+			return ctrl.Result{}, err
+		}
+		// ServiceAccount created successfully - return and requeue
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
I think the reconciler should not requeue here, but continute the reconciliation